tests/test_models.py

```diff
@@ -43,10 +44,132 @@ def tearDown(self):
         db.session.remove()
         db.drop_all()
 
+######################################################################
+#  H E L P E R   M E T H O D S
+######################################################################
+
+    def _create_wishlist(self, items=[]):
+        """ Creates an account from a Factory """
+        fake_wishlist = WishlistFactory()
+        wishlist = Wishlist(
+            name=fake_wishlist.name, 
+            user_id=fake_wishlist.user_id,
+            items=items
+        )
+        self.assertTrue(wishlist is not None)
+        self.assertEqual(wishlist.id, None)
+        return wishlist
+
+    def _create_item(self):
+        """ Creates fake items from factory """
+        fake_item = ItemFactory()
+        item = Item(
+            wishlist_id=fake_item.wishlist_id,
+            product_name=fake_item.product_name,
+            product_id=fake_item.product_id
+        )
+        self.assertTrue(item is not None)
+        self.assertEqual(item.id, None)
+        return item
+
 ######################################################################
 #  T E S T   C A S E S
 ######################################################################
 
+    def test_create_a_wishlist(self):
+        """ Create a wishlist and assert that it exists """
+        fake_wishlist = WishlistFactory()
+        wishlist = Wishlist(
+            name=fake_wishlist.name, 
+            user_id=fake_wishlist.user_id
+        )
+        self.assertTrue(wishlist != None)
+        self.assertEqual(wishlist.id, None)
+        self.assertEqual(wishlist.name, fake_wishlist.name)
+        self.assertEqual(wishlist.user_id, fake_wishlist.user_id)
+
+    def test_add_a_wishlist(self):
+        """ Create a wishlist and add it to the database """
+        wishlists = Wishlist.all()
+        self.assertEqual(wishlists, [])
+        wishlist = self._create_wishlist()
+        wishlist.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertEqual(wishlist.id, 1)
+        wishlists = Wishlist.all()
+        self.assertEqual(len(wishlists), 1)
+
+    def test_update_wishlist(self):
+        """ Update a wishlist """
+        wishlist = self._create_wishlist()
+        wishlist.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertEqual(wishlist.id, 1)
+
+        # Fetch it back
+        wishlist = Wishlist.find_or_404(wishlist.id)
```

Comment:
If this throws 404, the test fails with Exception and not Assertion. Same for rest.

Reply:
I don't catch your point. The test is fine. Just download the code and check. Sam for rest.

Reply:
This test looks good to me. This should not throw 404. If it does, it means that our code has a problem. That's the purpose of unit tests.